### DISCUSSION TITLE
Enable Calendar component to be provided nullable minDate and/or maxDate

### DISCUSSION THREAD

**Discussion by Cr3aHal0** [primefaces/community#1136] [PrimeNG] [answered]
For context : actually minDate & maxDate are 2 inputs of calendar that only accept non-nullable Date value. What it means is that you can have no value for any of them at startup but once you set any value (for example for search purpose) you can't reset it programmatically, i.e. providing undefined or null value. The only way to perform such reset is by using clear option (with [showClear]=true) or emptying the field manually (such leading in false value).

This change is all about enabling minDate & maxDate value to accept undefined value allowing free programmatical change, all checks about its truthyness being already handled. Let me know if null should be accepted instead/aswell.

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

Thanks for the feedback, it makes sense to accept nullable values like the other input components such as autocomplete / multiselect / dropdown. It's a minimal enhancement but highly useful for better DX. I understand your point, if possible, could you please provide a stackblitz example with simple guidance so we can target the issue precisely?

**Reply by Cr3aHal0** [reply]
Hi,

You can find the following PR about the change (https://github.com/primefaces/primeng/pull/14754), i'm trying to make a stackblitz with an example but the complex thing about the non nullable Date inputs is that it does not necessarily fail when serving apps (it  actually works) but rather at build time (see screenshot) with the strict type check.

![Sans titre-1](https://github.com/primefaces/community/assets/4007154/902ef742-fb80-436e-9b08-cfe54a0745f6)

Thank you for your quick reply
